version: "3.7"

volumes:
  redis_custom_cian:
    external: true
  postgres_custom_cian:
    external: true

services:
  redis:
    container_name: redis_custom_cian
    image: redis:alpine
    env_file:
      - ./.env.dev
    command: redis-server --requirepass sOmE_sEcUrE_pAsS
    ports:
        - 6379:6379
    volumes:
        - redis_custom_cian:/var/lib/redis
        - ${PWD}/conf/redis.conf:/usr/local/etc/redis/redis.conf

  database:
    container_name: postgres_custom_cian
    image: postgres:latest
    ports:
      - 5432:5432
    volumes:
      - postgres_custom_cian:/var/lib/postgresql/data
    env_file:
      - ./.env.dev

  web:
    build:
      context: ./custom_cian
      dockerfile: Dockerfile
    volumes:
      - ./custom_cian:/usr/src/custom_cian/
    command: >
      sh -c "python manage.py migrate --noinput &&
             python manage.py wait_for_db &&
             python manage.py runserver 0.0.0.0:8000"
    ports:
      - 8000:8000
    env_file:
      - ./.env.dev
    depends_on:
      - database
      - redis
    links:
      - database
      - redis

  celery:
    restart: always
    build:
      context: ./custom_cian
    command: celery -A custom_cian worker -l info
    volumes:
      - ./custom_cian:/usr/src/custom_cian
    env_file:
      - ./.env.dev
    depends_on:
      - database
      - redis
      - web
